src/main/java/mate/academy/springboot/datajpa/controller/CategoryController.java

```diff
@@ -0,0 +1,58 @@
+package mate.academy.springboot.datajpa.controller;
+
+import mate.academy.springboot.datajpa.dto.CategoryRequestDto;
+import mate.academy.springboot.datajpa.dto.CategoryResponseDto;
+import mate.academy.springboot.datajpa.dto.mapper.CategoryMapper;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/categories")
+public class CategoryController {
+    private CategoryService categoryService;
+    private CategoryMapper categoryMapper;
+
+    public CategoryController(CategoryService categoryService, CategoryMapper categoryMapper) {
+        this.categoryService = categoryService;
+        this.categoryMapper = categoryMapper;
+    }
+
+    @PostMapping("/add")
```

Comment:
```suggestion
    @PostMapping
```
this is redundant if you follow the REST way, you need to do a creational request like this `POST: /categories`